packages/eas-cli/src/commands/build/index.ts

```diff
@@ -208,3 +217,71 @@ export default class Build extends EasCommand {
     };
   }
 }
+
+async function handleDeprecatedEasJsonAsync(
+  projectDir: string,
+  nonInteractive: boolean
+): Promise<void> {
+  const reader = new EasJsonReader(projectDir);
+  const profileNames = await reader.getBuildProfileNamesAsync();
+  const platformAndProfileNames: [Platform, string][] = profileNames.flatMap(profileName => [
+    [Platform.ANDROID, profileName],
+    [Platform.IOS, profileName],
+  ]);
+
+  const deprecatedProfiles: [Platform, string][] = [];
+
+  for (const [platform, profileName] of platformAndProfileNames) {
+    const buildProfile = await reader.getBuildProfileAsync(platform, profileName);
+    if (buildProfile.artifactPath) {
+      deprecatedProfiles.push([platform, profileName]);
+    }
+  }
+
+  if (deprecatedProfiles.length === 0) {
+    return;
+  }
+
+  const deprecatedProfileNames = uniq(deprecatedProfiles.map(([, profileName]) => profileName));
+  Log.warn(`Some of your build profiles use deprecated ${chalk.bold('artifactPath')} field:`);
+  for (const profileName of deprecatedProfileNames) {
+    Log.warn(`- ${profileName}`);
+  }
+  Log.newLine();
+
+  if (nonInteractive) {
+    Log.warn(
+      `${figures.warning} Action required: rename ${chalk.bold('artifactPath')} to ${chalk.bold(
+        'applicationArchivePath'
+      )} in all of the build profiles listed above.`
+    );
+    Log.warn(
+      `See ${link('https://docs.expo.dev/build-reference/eas-json/')} for more information.`
+    );
+    Log.warn(
+      'This warning will become an error in an upcoming EAS CLI release. For now, we will proceed to avoid disruption of your builds.'
+    );
+    return;
+  }
+
+  const rename = await selectAsync('Do you want to handle renaming the field for you?', [
```

Comment:
```suggestion
  const rename = await selectAsync('Do you want EAS CLI to handle renaming the field for you?', [
```
Or something similar.

Reply:
```suggestion
  const rename = await selectAsync('Do you want us to handle renaming the field for you?', [
```